src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -81,10 +88,111 @@ master_create_distributed_table(PG_FUNCTION_ARGS)
 	text *distributionColumnText = PG_GETARG_TEXT_P(1);
 	Oid distributionMethodOid = PG_GETARG_OID(2);
 
+	ConvertToDistributedTable(distributedRelationId, distributionColumnText,
+							  distributionMethodOid, INVALID_COLOCATION_ID);
+	PG_RETURN_VOID();
+}
+
+
+/*
+ * create_distributed_table accepts a table, distribution column and
+ * distribution method, then it creates a distributed table.
+ */
+Datum
+create_distributed_table(PG_FUNCTION_ARGS)
+{
+	Oid relationId = PG_GETARG_OID(0);
+	text *distributionColumnText = PG_GETARG_TEXT_P(1);
+	Oid distributionMethodOid = PG_GETARG_OID(2);
+
+	Relation distributedRelation = NULL;
+	Relation pgDistColocation = NULL;
+	Node *distributionKey = NULL;
+	Var *distributionColumn = NULL;
+	char *distributionColumnName = NULL;
+	int distributionColumnType = 0;
+	uint32 colocationId = INVALID_COLOCATION_ID;
+
+	/* if distribution method is not hash, just create partition metadata */
+	char distributionMethod = LookupDistributionMethod(distributionMethodOid);
+	if (distributionMethod != DISTRIBUTE_BY_HASH)
+	{
+		ConvertToDistributedTable(relationId, distributionColumnText,
+								  distributionMethodOid, INVALID_COLOCATION_ID);
+		PG_RETURN_VOID();
+	}
+
+	/* get distribution column type */
+	distributionColumnName = text_to_cstring(distributionColumnText);
+	distributedRelation = relation_open(relationId, AccessShareLock);
+	distributionKey = BuildDistributionKeyFromColumnName(distributedRelation,
+														 distributionColumnName);
+	distributionColumn = (Var *) distributionKey;
+	distributionColumnType = distributionColumn->vartype;
+
+	/*
+	 * Get an exclusive lock on the colocation system catalog. Therefore, we
+	 * can be sure that there will no modifications on the table until this
+	 * transaction is committed.
+	 */
+	pgDistColocation = heap_open(DistColocationRelationId(), ExclusiveLock);
+
+	/* check for existing colocations */
+	colocationId = ColocationId(ShardCount, ShardReplicationFactor,
+								distributionColumnType);
+
+	/*
+	 * If there is a colocation group for the current configuration, get a
+	 * colocated table from the group and use its shards as a reference to
+	 * create new shards. Otherwise, create a new colocation group and create
+	 * shards with the default round robin algorithm.
+	 */
+	if (colocationId != INVALID_COLOCATION_ID)
+	{
+		Oid colocatedTableId = ColocatedTableId(colocationId);
+		ConvertToDistributedTable(relationId, distributionColumnText,
+								  distributionMethodOid, colocationId);
+
+		CreateColocatedShards(relationId, colocatedTableId);
+	}
+	else
+	{
+		colocationId = GetNextColocationId();
+		ConvertToDistributedTable(relationId, distributionColumnText,
+								  distributionMethodOid, colocationId);
+		CreateColocationGroup(colocationId, ShardCount, ShardReplicationFactor,
+							  distributionColumnType);
+
+		/* use the default way to create shards */
+		CreateRoundRobinShards(relationId, ShardCount, ShardReplicationFactor);
+	}
+
```

Comment:
This table would be colocated with lots of tables, so printing all of them would be a bit overkill.